Added the turkish localization

OsmanBuyuksar opened the pull request:
Solution for this issue: https://github.com/goatshriek/stumpless/issues/388#issue-1971501202

Implemented turkish localization as described [here](https://github.com/goatshriek/stumpless/blob/latest/docs/localization.md)

Commit:
Added the turkish localization

goatshriek commented on CMakeLists.txt:
```diff
@@ -210,6 +210,9 @@ elseif(LOCALE MATCHES "^da[_-]")
   set(USE_LOCALE_HE_IL TRUE)
+elseif(LOCALE MATCHES "^tr[_-]")
+  set(STUMPLESS_LANGUAGE "tr-TR")
+  set(USE_LOCALE_HE_IL TRUE)
```
This should instead be `USE_LOCALE_TR_TR`.

goatshriek reviewed (changes requested):
Thank you very much for contributing this! Just two changes to make, the first listed below. The second is to add a new section to the `.github/workflows/locale.yml` for the new translation. Commit 5529e077eb75c6e79d5634e9d7fc708828b78f89 has a good example of how to make this change if you need something to look at for comparison.

Commit:
Fix Locale in CmakeLists and add the tr section to .github/workflows/locale.yml

OsmanBuyuksar commented:
Sry about that, i have 2 repos to keep the contributing repo clean and looks like i made some of the changes to other repo :D. Anyways i fixed locale in cmakelists and also added the tr section to locale.yml file. Let me know if there is anything else.

goatshriek commented:
Looks like there's just one more thing to check - at least one line is longer than 80 characters. Once you've broken that up this should be all good to go.

Commit:
Fix row longer than 80 characters

OsmanBuyuksar commented:
There shouldn't be any line longer than 80 characters now, is there anything else?
